Detectors/TPC/workflow/src/CalibdEdxSpec.cxx

```diff
@@ -0,0 +1,132 @@
+    LOG(INFO) << "Created " << infoVec.size() << " objects for TF " << tfcounter;
+  }
+
+  void endOfStream(EndOfStreamContext& eos) final
+  {
+    LOG(INFO) << "Finalizing calibration";
+    // FIXME: not sure about this
+    constexpr uint64_t INFINITE_TF = 0xffffffffffffffff;
+    mCalibrator->checkSlotsToFinalize(INFINITE_TF);
+    sendOutput(eos.outputs());
+    mCalibrator->dumpToFile("mip_position.root");
+  }
+
+ private:
+  void sendOutput(DataAllocator& output)
+  {
+    // extract CCDB infos and calibration objects, convert it to TMemFile and send them to the output
+    // TODO in principle, this routine is generic, can be moved to Utils.h
+    using clbUtils = o2::calibration::Utils;
+    const auto& payloadVec = mCalibrator->getMIPVector();
+    auto& infoVec = mCalibrator->getInfoVector(); // use non-const version as we update it
+    assert(payloadVec.size() == infoVec.size());
+
+    // FIXME: not sure about this
+    for (uint32_t i = 0; i < payloadVec.size(); i++) {
+      auto& entry = infoVec[i];
+      auto image = o2::ccdb::CcdbApi::createObjectImage(&payloadVec[i], &entry);
+      LOG(INFO) << "Sending object " << entry.getPath() << "/" << entry.getFileName() << " of size " << image->size()
+                << " bytes, valid for " << entry.getStartValidityTimestamp() << " : " << entry.getEndValidityTimestamp();
+      output.snapshot(Output{o2::calibration::Utils::gDataOriginCDBPayload, "TPC_MIPposition", i}, *image.get()); // vector<char>
+      output.snapshot(Output{o2::calibration::Utils::gDataOriginCDBWrapper, "TPC_MIPposition", i}, entry);        // root-serialized
+    }
+    if (payloadVec.size()) {
+      mCalibrator->initOutput(); // reset the outputs once they are already sent
+    }
+  }
+
+  std::unique_ptr<CalibratordEdx> mCalibrator;
+};
+
+DataProcessorSpec getCalibdEdxSpec()
+{
+  using device = o2::tpc::CalibdEdxDevice;
+  using clbUtils = o2::calibration::Utils;
+
+  std::vector<OutputSpec> outputs;
+
+  // FIXME: not sure about this
+  outputs.emplace_back(ConcreteDataTypeMatcher{o2::calibration::Utils::gDataOriginCDBPayload, "TPC_MIPposition"});
+  outputs.emplace_back(ConcreteDataTypeMatcher{o2::calibration::Utils::gDataOriginCDBWrapper, "TPC_MIPposition"});
+
+  return DataProcessorSpec{
+    "tpc-calib-dEdx",
+    // select("tracks:TPC/TRACKS"),
+    Inputs{
+      InputSpec{"tracks", "TPC", "TRACKS"},
+    },
+    outputs,
+    adaptFromTask<device>(),
+    Options{
+      {"tf-per-slot", VariantType::Int, 100, {"number of TFs per calibration time slot"}},
+      {"max-delay", VariantType::Int, 3, {"number of slots in past to consider"}},
+      {"min-entries", VariantType::Int, 100, {"minimum number of entries to fit single time slot"}},
+      {"min-momentum", VariantType::Double, 0.4, {"minimum momentum cut"}},
+      {"max-momentum", VariantType::Double, 0.6, {"maximum momentum cut"}},
+      {"min-clusters", VariantType::Int, 60, {"minimum number of clusters in a track"}},
```

Comment:
at some point we should put this into a track cut class. For the moment it is fine.